src/System.Diagnostics.Process/tests/ProcessTests.Unix.cs

```diff
@@ -304,9 +304,93 @@ public void TestStartOnOSXWithBadFormat()
             }
         }
 
+        [Fact]
+        public void TestStartWithNonExistingUserThrows()
+        {
+            Process p = CreateProcessPortable(RemotelyInvokable.Dummy);
+            p.StartInfo.UserName = "DoesNotExist";
+            Assert.Throws<Win32Exception>(() => p.Start());
+        }
+
+        /// <summary>
+        /// Tests when running as a normal user and starting a new process as the same user
+        /// works as expected.
+        /// </summary>
+        [Fact]
+        public void TestStartWithNormalUser()
+        {
+            TestStartWithUserName();
+        }
+
+        /// <summary>
+        /// Tests when running as root and starting a new process as a normal user,
+        /// the new process doesn't have elevated privileges.
+        /// </summary>
+        [Fact]
+        [OuterLoop("Needs sudo access")]
+        [Trait(XunitConstants.Category, XunitConstants.RequiresElevation)]
+        public void TestStartWithRootUser()
+        {
+            RemoteInvokeOptions options = new RemoteInvokeOptions()
+            {
+                Start = false,
+                RunAsSudo = true
+            };
+            Process p = RemoteInvoke((Func<int>)TestStartWithUserName, options).Process;
+            AddProcessForDispose(p);
+
+            p.Start();
+            p.WaitForExit();
+
+            Assert.Equal(0, p.ExitCode);
+        }
+
+        public static int TestStartWithUserName()
+        {
+            using (ProcessTests testObject = new ProcessTests())
+            {
+                string realUserName;
+                if (geteuid() == 0)
+                {
+                    realUserName = Environment.GetEnvironmentVariable("SUDO_USER");
+                }
+                else
+                {
+                    realUserName = Environment.UserName;
+                }
+
+                Assert.NotNull(realUserName);
+                Assert.NotEqual("root", realUserName);
+
+                using (Process p = testObject.CreateProcessPortable(TestStartWithUserName_ChildProcess))
+                {
+                    p.StartInfo.UserName = realUserName;
+                    Assert.True(p.Start());
+
+                    p.WaitForExit();
+
+                    // since the process was started with the current real user, even if this test
+                    // was run with 'sudo', the child process will be run as the normal real user.
+                    // Assert that the effective user of the child process was never 'root'
+                    // and was the real user of this process.
+                    Assert.NotEqual(0, p.ExitCode);
+                }
+
+                return 0;
+            }
+        }
+
+        public static int TestStartWithUserName_ChildProcess()
```

Comment:
I'm confused by this function and how the name relates to its implementation.

Reply:
It enables the assert on line 376.

Reply:
This is the function that runs in the child process.  I guess I was naming it based on where its used.

I can rename it to something that more matches what it does - like `GetCurrentEffectiveUserId()`.  Would that help?